io-engine/src/grpc/v0/mayastor_grpc.rs

```diff
@@ -249,6 +249,9 @@ impl From<LvsError> for tonic::Status {
                 Errno::EMEDIUMTYPE => Status::aborted(e.to_string()),
                 _ => Status::internal(e.to_string()),
             },
+            LvsError::RepResize {
+                ..
+            } => Status::internal("replica resize failed".to_string()),
```

Comment:
Shouldn't we inspect the errno code rather than harcode to internal?
Do we get different error codes from spdk depending on error condition, if new size is set to larger than pool etc?

Reply:
> Shouldn't we inspect the errno code rather than harcode to internal? Do we get different error codes from spdk depending on error condition, if new size is set to larger than pool etc?

This seems like a reasonable mapping based on current possible spdk codes?
  
>   LvsError::RepResize {
                source, ..
            } => match source {
                Errno::ENOSPC | Errno::ENOMEM => Status::failed_precondition("Out of space/memory: replica resize failed".to_string()),
                Errno::EPERM => Status::permission_denied("operation not allowed: replica resize failed".to_string()),
                Errno::EINVAL => Status::failed_precondition("lvol missing? replica resize failed".to_string()),
                _ => Status::internal("internal error: replica resize failed".to_string())
            },

Reply:
I guess so, you may need to inspect the spdk code to check which error codes may be returned and how they should be converted to gRPC status codes